homeassistant/components/deconz/config_flow.py

```diff
@@ -254,9 +254,15 @@ async def async_step_hassio(self, discovery_info: HassioServiceInfo) -> FlowResu
         self.bridge_id = normalize_bridge_id(discovery_info.config[CONF_SERIAL])
         await self.async_set_unique_id(self.bridge_id)
 
-        self.host = discovery_info.config[CONF_HOST]
-        self.port = discovery_info.config[CONF_PORT]
-        self.api_key = discovery_info.config[CONF_API_KEY]
+        self.host: str = discovery_info.config[CONF_HOST]
+        self.port: int = discovery_info.config[CONF_PORT]
+        self.api_key: str = discovery_info.config[CONF_API_KEY]
+
+        # Workaround for https://github.com/home-assistant/addons/issues/2831
+        for entry_id in self.hass.data.get(DOMAIN, []):
+            gateway: DeconzGateway = self.hass.data[DOMAIN][entry_id]
+            if gateway.bridgeid == self.bridge_id and gateway.available:
+                return self.async_abort(reason="already_configured")
```

Comment:
This makes no sense at all, nothing warrants that the entry is already set up. As a matter of fact, the discovery can be done sooner than setting up the entry, which will still trigger this.

So, this is not a workaround or a fix.

Reply:
Check the referenced issue. That's what the logs show